src/Uri.php

```diff
@@ -236,8 +237,11 @@ public function withScheme($scheme): UriInterface
      *
      * {@inheritdoc}
      */
-    public function withUserInfo($user, $password = null): UriInterface
-    {
+    public function withUserInfo(
+        $user,
+        #[SensitiveParameter]
```

Comment:
This does indeed look like a bug in PHPCS.

Disabling the rule for this file could suffice.